[Adib Rifai] iP

bigismols opened the pull request:
#WansBot :
Have you ever wanted any of the following?
* daily task manager
* A ~~not so understanding or functioning~~ friend
* **really cool _ROBOT_**

Then look no further! Click [here](https://github.com/bigismols/ip.git) then,

1. Download the bot
2. Open the terminal by opening your search bar, typing cmd and pressing enter
3. Go to the directory using the following command with something like this 'cd path/to/directory_containing_wansbot'
4. Run the following command
```
java -jar wansbot.jar
```
5. Enjoy your **AWESOME** new chatbot!

> [!CAUTION]
> WansBot is still in the development process so please be kind 🥇 

WansBot Features:
- [x] Task Tracking
- [x] Managing Deadlines
- [ ] Reminders (maybe coming soon)

If you run into any issues, just remember,
> "It's not a bug, it's an undocumented feature"

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bigismols committed:
Changed chatbot name, added skeleton for the next steps, goodbye on exit

bigismols committed:
changed bot messages

bigismols committed:
Added the echo function and changed some bot messages

bigismols committed:
Cleaned up code

bigismols committed:
Added task and tasklist classes to src

bigismols committed:
Added adding tasks to task list function

bigismols committed:
Completed list and add function!

bigismols committed:
Added skeleton for todos and deadlines

bigismols committed:
added finish and unfinish to tasklist and tasks

bigismols committed:
added mark and unmark functionality for user

bigismols committed:
Updated todos and deadlines formatting

bigismols committed:
added events

bigismols committed:
cleaned up logic for level-4

bigismols committed:
configured testing for text UIs

bigismols committed:
added new empty error exception file

bigismols committed:
added rmpty input exception and its logic in bot class

bigismols committed:
logic for wrong inputs for mark and unmark implemented

bigismols committed:
Implemented exception for deadline command

bigismols committed:
Implemented logic for exception handling for event and deadline

bigismols committed:
Added removeTask method to TaskList

bigismols committed:
updated wansbot remove functionality

bigismols committed:
Remove exception handling added to WansBot

bigismols committed:
Add private static for WansBot methods

bigismols committed:
create package tasks

bigismols committed:
encapsulate bot function into methods

bigismols committed:
use switch statements instead of if else

bigismols committed:
implement writing task to ./data/tasklist.txt

bigismols committed:
add functionality to load tasks

bigismols committed:
implement loading and saving tasks to and from storage

bigismols committed:
fix bot output when loadTask()

bigismols committed:
add description comments before each method

bigismols committed:
add LocalDate functionality for deadline and event commands

bigismols committed:
add LocalDate function to load and save tasks

bigismols committed:
package classes and add find task date funcitonality

bigismols committed:
add description before each function

bigismols committed:
fix format of import statements

bigismols committed:
add UI package

bigismols committed:
add more functionality to UI class and delete redundant numTasks++ operation

bigismols committed:
implement al ui commands in UI package

bigismols committed:
add storage package and implement its functionality in Wansbot

bigismols committed:
Merge branch 'add-gradle-support'

bigismols committed:
fix gradle and storage so that ./gradelew can be run

bigismols committed:
add gradle checkstyle folder from addressbook level3

bigismols committed:
configure build.gradle

bigismols committed:
fix styling to coding standard with checkstyle for storage

bigismols committed:
fix other formatting errors (except for missing javadoc comments) with checkstyle

bigismols committed:
add junit test for number method in TaskList

bigismols committed:
add test for emptyInput

bigismols committed:
add JavaDoc comments for WansBot and UI packages

bigismols committed:
add JavaDoc comments for all but storage package.

bigismols committed:
add JavaDoc comments for storage and fix all style using checkstyle.

bigismols committed:
fix coding style to coding standard for java

bigismols committed:
add function findKeyword

bigismols committed:
Merge tag 'A-CodingStandard'

bigismols committed:
Merge branch 'branch-level-9'

# Conflicts: 0
#	src/main/java/wansbot/ui/UI.java

bigismols committed:
add JavaDoc for findKeyword

wrjgold commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,287 @@
+    /**
+     * Throws InputEmptyException when user doesn't input anything after command word.
+     */
+    protected static void emptyInput(String userInput) throws InputEmptyException {
```
Maybe using a verb to describe this method will make it clearer? For e.g. checkEmptyInput

bigismols replied:
True, thanks for spotting that.

wrjgold commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,287 @@
+     * Throws NumberFormatException and NotANumMarkingException when a non-number is inputted and when it's not
+     * a valid task number in the userTaskList.
+     */
+    private static void notNumInput(String userInput, int taskListSize) throws NumberFormatException,
```
Similarly, using a verb like checkNotNumInput will be clearer :)

bigismols replied:
Yep, you're right on that one

wrjgold commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,287 @@
+     * Handles marking tasks function by specifying how caught exceptions are dealt with. Prints to console what
+     * user needs to correct for the command to work.
+     */
+    private static void markTasks(String userInput) {
```
Making "Tasks" plural might confuse some users thinking they can pass in a list of Task indexes

bigismols replied:
Oh yes, I should've thought of that. Thanks for pointing that out!

wrjgold commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,287 @@
+    /**
+     * Handles unmarking tasks function by specifying to user what went wrong.
+     */
+    private static void unmarkTasks(String userInput) {
```
Similar to above, consider making Tasks singular

bigismols replied:
Yep yep.

wrjgold commented on src/main/java/wansbot/ui/UI.java:
```diff
@@ -0,0 +1,247 @@
+                + "Hey, I'm\n"
+                + logo
+                + "\nCan I help? (I can only manage a todo list so...)\n" + HR);
+    }
```
I like how you gave your bot more personality

bigismols replied:
Just trying to make this long exercise a little more fun. :o

wrjgold reviewed (commented):
Generally well written code! However can consider some better use of naming. For e.g verbs to describe the methods. Do also watch out for singular/plural nouns which can confuse users

bigismols committed:
style: changed style of Task.java according to checkstyle

bigismols committed:
style: changed style according to checkstyle in UI,java

bigismols committed:
feat: add javaFX configuration to build.gradle

bigismols committed:
feat: add skeleton for javaFX

bigismols committed:
feat: add fxml as gui without chatbot function

bigismols committed:
feat: add mark, unmark and list funcitonality to bot

bigismols committed:
feat: add rest of functionailities in wansbot

bigismols committed:
feat: add more styling to GUI

bryanjhc commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,337 @@
+     * Throws custom InputEmptyException for events when either missing from date after /from, missing to date after
+     * /to or missing both.
+     */
+    protected static void missingInputEvent(String userInput) {
```
Maybe we could use a verb to name this method similar to @wrjgold's comment (eg. checkMissingInputEvent)

bigismols replied:
Yes sir, you're right.

bryanjhc commented on src/main/java/wansbot/WansBot.java:
```diff
@@ -0,0 +1,337 @@
+    /**
+     * Adds Deadlined to userTaskList. If there is invalid input, WansBot will guide user on the correct commands.
+     */
+    private String addDeadlined(String userInput) {
```
did you maybe mean to write addDeadlines here?

bigismols replied:
Actually, because Deadlined is the name of my class, I intentionally made it that way. Maybe I should refactor it to Deadline to make it clearer

bryanjhc commented on src/main/java/wansbot/tasks/Deadlined.java:
```diff
@@ -0,0 +1,47 @@
+     * @param date is a LocalDate input by user as part of filtering tasks.
+     * @return Boolean which tells user if the Deadline is by date.
+     */
+    public boolean isOnDate(LocalDate date) {
```
Good job using proper naming for booleans!

bigismols replied:
Thanks bro!

bryanjhc reviewed (commented):
LGTM boss! just a few nits here and there!

bigismols committed:
Hello "command": make WansBot say hello on initialisation

WansBot does not say hello on initialisation

Very peculiar for user to start up the bot and end up not being greeted by the bot. User might feel a little lost then.

Let's add a DialogBox with WansBot's response to a 'hello' command on under the method setWansBot.

bigismols committed:
Assertions: add assertions.

No assertions implemented in any part of the code.

Assertions allows developers to test invariants and assumptions in their code. When assertion fails we can know where the error occured.

Let's add assertions where there are class invariants, post conditions and any internal invariants

bigismols committed:
style: fix code quality for storage class

bigismols committed:
style: fix code quality of WansBot

bigismols committed:
feat: add new package questions.

bigismols committed:
feat: add functionality to remember and ask user questions

bigismols committed:
style: add javaDoc comments for QuestionBank

bigismols committed:
stye: add javaDoc comments to new WansBot and Question methods and ui Methods

bigismols committed:
Merge pull request #1 from bigismols/heads/A-Assertions

Heads/a assertions

bigismols committed:
Merge branch 'master' into A-CodeQuality

bigismols committed:
Merge branch 'master' into D-Trivia

bigismols committed:
Merge pull request #2 from bigismols/D-Trivia

D trivia

bigismols committed:
Merge branch 'master' into A-CodeQuality

bigismols committed:
Merge pull request #3 from bigismols/A-CodeQuality

A code quality

bigismols committed:
feat: add css folders and some styling

bigismols committed:
feat: style UI to make it easier to the user's eyes

bigismols committed:
feat: add red border when user makes error in command

bigismols committed:
fix: test method name wrong, corrected

bigismols committed:
feat: add new test to WansBotTest

bigismols committed:
feat: add more tests to WansBotTest

bigismols committed:
feat: add error handling when addEvent /from date is after the /to date

bigismols committed:
feat: add UI screenshot to docs folder

bigismols committed:
fix: change png name to Ui from UI

bigismols committed:
fix: replace Ui.png with corrected Ui.png

bigismols committed:
fix: unreachable findname and findtask command

bigismols committed:
Update README.md

bigismols committed:
fix: findname and find taskdate commands

bigismols committed:
Merge branch 'master' of https://github.com/bigismols/ip

bigismols committed:
Update README.md

bigismols committed:
Rename README.md to docs/READme.md

bigismols committed:
Delete docs/READme.md

bigismols committed:
Update README.md

bigismols committed:
fix: fix bug with question wrong format

bigismols committed:
fix: detect bug with todos format

bigismols committed:
fix: fix bug with detecting wrong answer command format

bigismols committed:
fix: detect format bug for findname command

bigismols committed:
fix: detect bug with the findtask command

bigismols committed:
fix: detect bug in remove, mark, unmark command

bigismols committed:
feat: add bot says bye function

bigismols committed:
feat: add icon and title to window.

bigismols committed:
feat: add new Ui screenshot to docs folder

bigismols committed:
Add javadoc comment for handle WrongMarkFormat

bigismols committed:
Add javadoc comment to getDukeDialogBox

bigismols committed:
Refactor getDukeDialogBox to getWansBotDialogBox for clarity

bigismols committed:
Add javadoc comment for checkValidDates

bigismols committed:
Extract recogniseQuestion method in QuestionBank class
